web-api/src/personalization/rerank.rs

```diff
@@ -81,8 +95,11 @@ pub(super) fn rerank_by_interest_and_tag_weight(
     let tag_weight_ranks = rerank_by_tag_weights(documents, tag_weights);
 
     for document in documents.iter_mut() {
-        document.score = interest_ranks[&document.id]
+        let personalization_score = interest_ranks[&document.id]
             .merge_as_score(tag_weight_ranks[&document.id], interest_tag_bias);
+
+        document.score = personalization_ratio * personalization_score
+            + (1. - personalization_ratio) * document.score;
     }
 
     let secondary_sorting_factor = if interest_tag_bias >= 0.5 {
```

Comment:
:question:  I just keep it the way it is, but it's not clear if we might want to change this depending on `persoanlization_bias`.

E.g. if personalization_bias < 0.5 sort by `(score, original_score)` and if it's `>= 0.5` sort by `(score, secondary_sorting_factor)` or similar.

Reply:
with the array of weights in mind, we can break ties by selecting the scores map related to the highest weight in the array.